### PR TITLE
chore: upgrade Streams and update tests. revert this after KIP-479

### PR DESCRIPTION
**Pull request by agavra**
### Description 
Will write description in a bit. Trigger build

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/EndToEndEngineTestUtil.java [hunk 1]
```diff
@@ -102,26 +106,30 @@ static void writeExpectedTopologyFiles(
...
+
+          writeExpectedTopologyFile(
+              testCase.name,
+              queryMetadata,
+              configsToPersist,
+              objectWriter,
+              topologyDir);
+        }
+      } catch (final Exception e) {
+        LOG.warn("Ignoring test case {} due to error {}", testCase, e);
```
Comment (rodesai) [inline ksql-functional-tests/src/test/java/io/confluent/ksql/test/EndToEndEngineTestUtil.java]: do we actually hit this?

Comment (agavra) [reply ksql-functional-tests/src/test/java/io/confluent/ksql/test/EndToEndEngineTestUtil.java]: Yeah - I hit it for the TOSTRUCT functionality. I meant to remove this and forgot, we should debug it not ignore it.

I'll back this out.

File: ksql-functional-tests/src/test/resources/expected_topology/5_3_0/sink-partitions-replicas_-_should_copy_partition_and_replica_count_from_source_topic [hunk 2]
```diff
@@ -15,14 +16,18 @@
...
   "ksql.streams.commit.interval.ms" : "2000",
   "ksql.streams.auto.commit.interval.ms" : "0",
   "ksql.sink.replicas" : null,
+  "ksql.metrics.extension" : null,
   "ksql.streams.topology.optimization" : "all",
   "ksql.streams.num.stream.threads" : "4",
+  "ksql.metrics.tags.custom" : "",
   "ksql.query.fields.key.legacy" : "false",
   "ksql.udfs.enabled" : "true",
+  "ksql.connect.registry.url" : "http://localhost:8083",
```
Comment (vcrfxia) [inline ksql-functional-tests/src/test/resources/expected_topology/5_3_0/sink-partitions-replicas_-_should_copy_partition_and_replica_count_from_source_topic]: What are these new configs for?

Comment (agavra) [reply ksql-functional-tests/src/test/resources/expected_topology/5_3_0/sink-partitions-replicas_-_should_copy_partition_and_replica_count_from_source_topic]: script generating issue... I'll back this out

**Comment by apurvam** [comment]
Do you mean revert this after [KIP-479](https://cwiki.apache.org/confluence/display/KAFKA/KIP-479%3A+Add+Materialized+to+Join)?

**Comment by agavra** [comment]
@apurvam - long week... thanks for catching that. I'll let the test run to make sure it passes and then I'll update all the comments and the commit name and merge.

**Pull request merged**